src/app/user-page/user-page.component.ts

```diff
@@ -56,30 +55,11 @@ export class UserPageComponent implements OnInit {
       },
       (error: HttpErrorResponse) => {
         this.alertService.detailedError(error);
+        this.router.navigateByUrl(''); //returns to homepage if user doesn't exist or another error occurs;
```

Comment:
Suggestion: I'd lean more towards displaying a PageNotFound instead of going to the home page.

1. You'd see the url with the username so you can maybe figure out what went wrong.
2. Sometimes you wonder what happened -- you clicked on username/bookmark and you're on a totally unexpected page.

But I could see an argument for just going to the home page. What do others think?

Reply:
I leaned more towards displaying a the Page Not Found page as well. But I kept it as the homepage since that's how it was set originally. Curious to see what others think as well!

Reply:
Changed the redirect to page not found